### DISCUSSION TITLE
Help with quiz 5 Q4

### DISCUSSION THREAD

**Discussion by Tania-Gardasevic** [pykale/transparentML#58] [Q&A] [answered]
Hello,

I have tried many times to get a correct answer for this Q but haven't been able to figure out what I've done wrong.

I ran the following and got an accuracy of 44.80 which is marked as wrong. Any help would be greatly appreciated!

```
data_url = "https://github.com/pykale/transparentML/raw/main/data/Smarket.csv"
df = pd.read_csv(data_url, na_values="?").dropna()
df["Direction"] = df["Direction"].factorize()[0]

np.random.seed(2022)
model=LogisticRegression(solver="newton-cg", penalty=None, max_iter=1000, random_state=2022)

X = df.drop(columns=["Today", "Direction"]).values
y = df["Direction"].values

X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=2022)
model.fit(X_train, y_train)
y_pred = model.predict(X_test)
accuracy = accuracy_score(y_test, y_pred)
print(f"Test Accuracy: {accuracy*100:.2f}%")
```

**Comment by Mdnaimulislam** [comment, marked as answer]
Hello @Tania-Gardasevic,

Thank you for pointing this out. We have reviewed the issue and found that recent updates to scikit-learn have affected the result in Chapter 5 (Q4). In earlier versions, LogisticRegression used penalty="none", whereas in newer versions this has been replaced with penalty=None. Previously, the expected answer for Chapter 5 (Q4) was 48.40, but with the library update, the correct answer is now 44.80. Therefore, your code and result are correct. For reference, the earlier result of 48.4 can be reproduced using scikit-learn==1.1.3; however, this version is not compatible with Python 3.12 or above.

We also identified that a few other quizzes were affected by the library version update, specifically Chapter 5 (Q5, Q6) and Chapter 10 (Q4). We have double-checked and updated all affected answers. Thank you again for bringing this to our attention. We encourage all to continue bringing any issues to our attention, and we are always here to support you.

Thanks,
@Mdnaimulislam
